components/suggest/src/db.rs

```diff
@@ -128,63 +128,72 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given query from the database.
     pub fn fetch_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
-        if let Some(suggestion) = self.fetch_yelp_suggestion(query)? {
-            return Ok(vec![suggestion]);
-        }
-
-        let keyword_lowercased = &query.keyword.to_lowercase();
-        let (keyword_prefix, keyword_suffix) = split_keyword(keyword_lowercased);
         let suggestions_limit = query.limit.unwrap_or(-1);
+        let suggestions =
+            query
+                .providers
+                .clone()
+                .into_iter()
+                .try_fold(vec![], |mut acc, provider| {
+                    let suggestions = match provider {
+                        SuggestionProvider::Amp => self.fetch_amp_suggestions(query),
+                        SuggestionProvider::Wikipedia => self.fetch_wikipedia_suggestions(query),
+                        SuggestionProvider::Amo => self.fetch_amo_suggestions(query),
+                        SuggestionProvider::Pocket => self.fetch_pocket_suggestions(query),
+                        SuggestionProvider::Yelp => {
+                            if let Some(yelp_suggestion) = self.fetch_yelp_suggestion(query)? {
+                                Ok(vec![yelp_suggestion])
+                            } else {
+                                Ok(vec![])
+                            }
```

Comment:
this felt jank

Reply:
Now we can make `fetch_yelp_suggestion()` return `Result<Vec<Suggestion>>` instead of `Result<Option<Suggestion>>`.

Reply:
Yeah, I like the idea of renaming `fetch_yelp_suggestion` to `fetch_yelp_suggestions`, and having it return a `Vec` instead of a single `Option`.